File: day2/day2.py
def password(i):
  count = 0

  for line in lines:
    temp = line.split()
    minmax = temp[0].split('-')
    minimum = int(minmax[0])
    maximum = int(minmax[1])
    letter = temp[1][0]
    password = temp[2]

    # print("min: " + minmax[0] + " max: " + minmax[1] + " letter: " + temp[1][0] + " password: " + temp[2])

    if minimum <= password.count(letter) <= maximum:
      count += 1

  return count

def password_two(i):
  count = 0

  for line in lines:
    temp = line.split()
    minmax = temp[0].split('-')
    i1 = int(minmax[0]) - 1
    i2 = int(minmax[1]) - 1
    letter = temp[1][0]
    password = temp[2]

    if (password[i1] == letter or password[i2] == letter) and password[i1] != password[i2]:
      count += 1
    
  return count


  
if __name__ == "__main__":
  with open('day2/day2input.txt', 'r') as f:
    lines = f.readlines()

  print(password(lines))
  print(password_two(lines))

    # print(line)

File: day2/day2.rb
def p1(i)
  c = 0

  i.each { |h| c += 1 if h[:minmax][0].to_i <= h[:password].count(h[:letter]) and h[:password].count(h[:letter]) <= h[:minmax][1].to_i }

  return c
end

def p2(i)
  c = 0

  i.each do |h|
    pw = h[:password]
    l = h[:letter]
    i1 = h[:minmax][0]
    i2 = h[:minmax][1]

    c += 1 if (pw[i1] == l or pw[i2] == l) and pw[i1] != pw[i2]
    
  end

  return c
end

if __FILE__ == $0
  data = []
  parsed = []
  File.foreach("day2/day2input.txt") { |line| data << line.chomp }
  
  data.each do |i|
    temp = i.split(" ")
    parsed << { 'minmax': temp[0].split('-'), 'letter': temp[1][0], 'password': temp[2] }
  end

  # puts parsed.to_s
  puts p1(parsed)
  puts p2(parsed)
end
